subscribe.go

```diff
@@ -32,7 +32,7 @@ type legSubscriber struct {
 	policy          PolicyHandler
 	defaultSelector ipld.Node
 
-	syncmtx    sync.Mutex
+	syncmtx    sync.RWMutex
```

Comment:
please document what this mutex locks; I imagine it's the two fields below, but I'm not 100% sure.

Reply:
By the way, a RWMutex has worse performance than a Mutex if we mostly just grab write locks - because it is a more complex mutex. So I would say it's best to leave this as a sync.Mutex, as only the tests need a read-only lock. Otherwise you might be making legSubscriber slightly slower in terms of lock contention.

Reply:
In other words, a RWMutex is only worthwhile if the majority of your locking operations are read-only. If the majority are read-write, then a Mutex will simply perform better.

Reply:
sounds good thanks Daniel